lib/matplotlib/mathtext.py

```diff
@@ -2873,25 +2873,37 @@ def accent(self, s, loc, toks):
                 ])
 
     def function(self, s, loc, toks):
-        self.push_state()
-        state = self.get_state()
-        state.font = 'rm'
-        hlist = Hlist([Char(c, state) for c in toks[0]])
-        self.pop_state()
+        hlist = self.operatorname(s, loc, toks)
         hlist.function_name = toks[0]
         return hlist
 
     def operatorname(self, s, loc, toks):
         self.push_state()
         state = self.get_state()
         state.font = 'rm'
+        hlist_list = []
         # Change the font of Chars, but leave Kerns alone
         for c in toks[0]:
             if isinstance(c, Char):
                 c.font = 'rm'
                 c._update_metrics()
+                hlist_list.append(c)
+            elif isinstance(c, str):
+                hlist_list.append(Char(c, state))
+            else:
+                hlist_list.append(c)
+        next_char_loc = loc + len(toks[0]) + 1
+        if isinstance(toks[0], ParseResults):
+            next_char_loc += 14  # Skip `operatorname{}`
+        next_char = next((c for c in s[next_char_loc:] if c != ' '), '')
+        delimiters = self._left_delim | self._ambi_delim | self._right_delim
+        delimiters |= set(r"^_".split())
```

Comment:
Is this right? `.split()` splits on whitespace, and there's none in this literal. Are you sure you don't want?
```suggestion
        delimiters |= {"^", "_"}
```

Reply:
You're correct. I broke it sometime between testing it and committing it, and I neglected to add a test case for it. I've now fixed it and added an appropriate unit test.